psi4/src/psi4/libfock/DirectJK.cc

```diff
@@ -677,29 +728,34 @@ void DirectJK::build_JK(std::vector<std::shared_ptr<TwoBodyAOInt>>& ints, std::v
                                 for (int q = 0; q < Qsize; q++) {
                                     for (int r = 0; r < Rsize; r++) {
                                         for (int s = 0; s < Ssize; s++) {
-                                            J1p[(p + Poff2) * dQsize + q + Qoff2] +=
-                                                prefactor * (Dp[r + Roff][s + Soff] + Dp[s + Soff][r + Roff]) *
-                                                (*buffer2);
-                                            J2p[(r + Roff2) * dSsize + s + Soff2] +=
-                                                prefactor * (Dp[p + Poff][q + Qoff] + Dp[q + Qoff][p + Poff]) *
-                                                (*buffer2);
-                                            K1p[(p + Poff2) * dRsize + r + Roff2] +=
-                                                prefactor * (Dp[q + Qoff][s + Soff]) * (*buffer2);
-                                            K2p[(p + Poff2) * dSsize + s + Soff2] +=
-                                                prefactor * (Dp[q + Qoff][r + Roff]) * (*buffer2);
-                                            K3p[(q + Qoff2) * dRsize + r + Roff2] +=
-                                                prefactor * (Dp[p + Poff][s + Soff]) * (*buffer2);
-                                            K4p[(q + Qoff2) * dSsize + s + Soff2] +=
-                                                prefactor * (Dp[p + Poff][r + Roff]) * (*buffer2);
-                                            if (!lr_symmetric_) {
-                                                K5p[(r + Roff2) * dPsize + p + Poff2] +=
-                                                    prefactor * (Dp[s + Soff][q + Qoff]) * (*buffer2);
-                                                K6p[(s + Soff2) * dPsize + p + Poff2] +=
-                                                    prefactor * (Dp[r + Roff][q + Qoff]) * (*buffer2);
-                                                K7p[(r + Roff2) * dQsize + q + Qoff2] +=
-                                                    prefactor * (Dp[s + Soff][p + Poff]) * (*buffer2);
-                                                K8p[(s + Soff2) * dQsize + q + Qoff2] +=
-                                                    prefactor * (Dp[r + Roff][p + Poff]) * (*buffer2);
+                                            if (build_J) {
```

Comment:
It's worth noting that if statements contained within tight inner loops like this can incur a non-negligible performance cost.

Reply:
Good point, though there is another if statement in the inner loop (lr_symmetric), so I think this is okay for now. Also, the conditional statements are within four O(1) loops, so the time would still be dominated by integral formation.